tools/jdbc/src/jni/duckdb_java.cpp

```diff
@@ -83,11 +83,27 @@ static jclass J_UUID;
 static jmethodID J_UUID_getMostSignificantBits;
 static jmethodID J_UUID_getLeastSignificantBits;
 
+static jclass J_DuckDBDate;
+static jmethodID J_DuckDBDate_getDaysSinceEpoch;
+
 void ThrowJNI(JNIEnv *env, const char *message) {
 	D_ASSERT(J_SQLException);
 	env->ThrowNew(J_SQLException, message);
 }
 
+static duckdb::vector<jclass> toFree;
+
+static jclass GetClassRef(JNIEnv *env, const string &name) {
```

Comment:
Thoughts on this @Mytherin? Right now we're not really cleaning up any of the global references, and I figure something like this makes it easier, plus it removes some boilerplate

Reply:
I have no concept of what a global reference in Java is, but I can guess :)

What I would suggest is we should create a `GlobalReferenceManager` singleton class that holds the references and correctly grabs locks. Perhaps they can even be cached similar to how the Python import caches work?